src/fluree/server/handler.clj

```diff
@@ -255,38 +256,74 @@
           fuel 1000] ; TODO: get this for real
       (assoc-in resp [:headers "x-fdb-fuel"] (str fuel)))))
 
-(defn wrap-policy-metadata
-  "Both Fluree-Policy-Class and Fluree-Policy-Identity can be optionally
-  set in the header for non-credentialed queries/transactions. The primary
-  motivation is enforcing policy with SPARQL which doesn't have an opts
-  map option, although if set, it will also override policyClass/Identity
-  that might have otherwise been set in FlureeQL/JSON."
+(defn wrap-header-opts
+  "Extract options from headers, parse and validate them where necessary, and attach to
+  request. Opts set in the header override those specified within the transaction or
+  query. Intended mainly to provide a method of specifying opts for SPARQL queries"
   [handler]
-  (fn [req]
-    (let [policy-identity (get-in req [:headers "fluree-policy-identity"])
-          policy-class    (get-in req [:headers "fluree-policy-class"])
-          policy          (when-let [p (get-in req [:headers "fluree-policy"])]
-                            (try
-                              (json/parse p false)
-                              (catch Exception _
-                                (throw (ex-info "Invalid Fluree-Policy header: must be JSON."
-                                                {:status 400})))))
-          policy-values   (when-let [pv (get-in req [:headers "fluree-policy-values"])]
-                            (try
-                              (let [pv* (json/parse pv false)]
-                                (if (sequential? pv*)
-                                  pv*
-                                  (throw (ex-info "Invalid Fluree-Policy-Values header, it must be a valid values binding: [[\"?varA\" \"?varB\"] [[<a1> <b1>] [<a2> <b2>] ...]]"
-                                                  {:status 400}))))
-                              (catch Exception _
-                                (throw (ex-info "Invalid Fluree-Policy-Values header: must be JSON."
-                                                {:status 400})))))]
-      (handler
-       (cond-> req
-         policy-identity (assoc :policy/identity policy-identity)
-         policy-class (assoc :policy/class policy-class)
-         policy (assoc :policy/policy policy)
-         policy-values (assoc :policy/values policy-values))))))
+  (fn [{:keys [headers credential/did] :as req}]
+    (let [{:keys [meta max-fuel identity policy-identity policy policy-class policy-values format output]}
+          (-> headers
+              (select-keys ["fluree-meta" "fluree-max-fuel" "fluree-identity" "fluree-policy-identity"
```

Comment:
I think these recognized keys should be defined in a separate top level binding so we can easily reference the recognized header options.